airflow/config_templates/config.yml

```diff
@@ -89,7 +89,7 @@ core:
         This defines the maximum number of task instances that can run concurrently per scheduler in
         Airflow, regardless of the worker count. Generally this value, multiplied by the number of
         schedulers in your cluster, is the maximum number of task instances with the running
-        state in the metadata database.
+        state in the metadata database. Setting this value to zero allows unlimited parallelism.
```

Comment:
Jus to clarify after the discussion in the issue.
What is the expected behavior in Airflow 3? Do we leave this as is or we wish to standardize this?
The issue mentions followup of additional docs so I'd rather to also leave todos in the code as reminder to cleanups we may wish to take for Airflow 3.

Reply:
Would the TODO be to remove support for this? Or to improve the docs even more (as in add something in the documentation outside of the config reference)?

Reply:
I am referencing https://github.com/apache/airflow/issues/41055#issuecomment-2253340939

My understanding that we add 0 support to be backward compatible but ideally we don't really want it. So the todo is to remove the 0 support in Airflow 3 and ask users to provide the limit explicitly.

Reply:
Yeah, I think we should remove it because:
1) It adds unnecessary complexity to the code
2) This behaviour can be achieved by setting parallelism to a sufficiently high number
3) (most importantly) I think it's actually important for the user to have to do 2), it forces them to actually think "hmm how parallel _should_ I actually run tasks? Is infinity appropriated? Will infinity actually cause degraded performance?". I think allowing 0 gives an easy way for folks to set and forget.

But I think I may be in the minority, so I did not add any TODOs to remove it (although I'd love to). So I suppose I'll wait to hear back from others on that front.

Reply:
I'm not against removing it. You can create an issue to remove this in Airflow 3 or create a draft PR. However, we shouldn't keep a TODO in the code (my suggestion).

Reply:
Okay, that sounds like a decent consensus. I'll merge this PR so that we have it for 2.10, and cut a ticket to propose it's removal in v3.

Reply:
#41162 41162